### ISSUE TITLE
[@lit/react] Add React 19 support to reduce migration burden

### ISSUE THREAD

**Issue by maxpatiiuk** [lit/lit#4857]
### Should this be an RFC?

- [x] This is not a substantial change

### Which package is this a feature request for?

React (@lit/react)

### Description

`@lit/react` appears to work in React 19 in my limited testing. The only issue is that this line in package.json needs to be updated to add `|| 19`

https://github.com/lit/lit/blob/267d243aeec13a6f0e8184420919db4c519b8caf/packages/react/package.json#L199

I understand that React 19 supports custom elements, removing the need for wrappers. However, to smooth the ecosystem migration, it might be a good idea to support wrappers in React 19 too.

Use cases:
- Forcing people to refactor ooutur all wrappers usages and drop `@lit/react` before adopting React 19 is going to make the migration burden harder. Better let people gradually adopt React 19 and drop wrappers afterward.
- In a monorepo, some packages might update to React 19 before others
  - Example: updating storybook may require updating to React 19, while the code is not ready to
- Using wrappers may have better DX because IDEs can auto-import them on usages, and detect needless imports (assuming the package that uses `@lit/react` has separate entrypoint for each component wrapper, with a side effect import of the component inside) - not so for side-effect component imports.

### Alternatives and Workarounds

Use yarn/npm "resolutions" to force `@llit/react` to work with `@types/react@19`.

**Comment by augustjk** [comment]
Indeed, we can and should widen the semver range for the types package.

We might even do better where we check `React.version` to see if it's >= 19 and pass all the props to React so that we don't have to do the imperative property setting in the `useLayoutEffect` for client rendering. This can prevent unwanted behavior where a component as its mounting momentarily has a default value for a property before it's set in the first layout effect.

We do have some special handling of element props for server rendering which we want to keep.

**Comment by OvidijusParsiunas** [comment]
Hi. Just wanted to double check if _this issue_ is now redundant since the [following](https://github.com/lit/lit/pull/4865) PR has now been merged and deployed in [@lit/react](https://www.npmjs.com/package/@lit/react) version `1.0.7`. Just wanted to double check if there was anything else left to be done.
